src/main/kotlin/com/demonwav/mcdev/i18n/I18nElementFactory.kt

```diff
@@ -0,0 +1,95 @@
+/*
+ * Minecraft Dev for IntelliJ
+ *
+ * https://minecraftdev.org
+ *
+ * Copyright (c) 2017 minecraft-dev
+ *
+ * MIT License
+ */
+
+package com.demonwav.mcdev.i18n
+
+import com.demonwav.mcdev.i18n.lang.I18nFile
+import com.demonwav.mcdev.i18n.lang.I18nFileType
+import com.demonwav.mcdev.i18n.lang.gen.psi.I18nEntry
+import com.demonwav.mcdev.util.applyWriteAction
+import com.intellij.ide.DataManager
+import com.intellij.openapi.actionSystem.DataContext
+import com.intellij.openapi.module.Module
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.ui.popup.JBPopupFactory
+import com.intellij.openapi.vfs.VirtualFile
+import com.intellij.psi.PsiElement
+import com.intellij.psi.PsiFileFactory
+import com.intellij.psi.PsiManager
+import com.intellij.psi.search.FileTypeIndex
+import com.intellij.psi.search.GlobalSearchScope
+import com.intellij.ui.components.JBList
+import com.intellij.util.Consumer
+import java.util.Locale
+
+object I18nElementFactory {
+    val DOMAIN_PATTERN = Regex("^.*?/assets/(.*?)/lang.*?\$")
```

Comment:
Mumfrey points out that (he thinks) the domain for Minecraft langs can only be a single name, rather than a directory structure, in which case the grouping is too greedy (should match `not` `/`). If he's wrong then this can be ignored.